python3/cloudwatch/src/lambda_function.py

```diff
@@ -88,11 +89,19 @@ def _parse_to_json(log):
 def _parse_cloudwatch_log(log, additional_data):
     # type: (dict, dict) -> bool
     _add_timestamp(log)
+
     if LAMBDA_LOG_GROUP in additional_data['logGroup']:
         if _is_valid_log(log):
             _extract_lambda_log_message(log)
         else:
             return False
+
+    if _has_whitelist():
+        return _is_whitelisted_log(log)
+
+    if _is_blacklisted_log(log):
+        return False
+
```

Comment:
ATM, if there's a whitelist it won't proceed to adding the additional data to the log or parse to json.

Reply:
Good catch, fixed.